lib/json-parse.js

```diff
@@ -1,35 +1,48 @@
+const RE2 = require('re2');
+
 module.exports = (txt, reviver, context = 20) => {
+  const regexToken = new RE2(/^Unexpected token.*position\s+(\d+)/i);
+  const regexEndof = new RE2(/^Unexpected end of JSON.*/i);
```

Comment:
Those regex can be extracted outside the function. It's unnecessary to re-compiled those const in every call of this function.